src/TagDependencyTrait.php

```diff
@@ -58,17 +73,27 @@ public function objectCompositeTag()
             return [];
         }
 
-        $cacheFields = (is_array($cacheFields) && !empty($cacheFields) && is_array($cacheFields[0])) ? $cacheFields : [$cacheFields];
         $tags = [];
 
         foreach ($cacheFields as $tagFields) {
             $tag = [];
+            $changed = false;
 
             foreach ($tagFields as $tagField) {
                 $tag[$tagField] = $this->$tagField;
+                $changed |= isset($oldfields[$tagField]);
```

Comment:
$oldfields should be defined as a function argument.

It will only have useful values from an update event.

Changed will only be true if one or more tagged object attributes is modified.

In that case changed should be true.